generator/parameter_test.go

```diff
@@ -1705,3 +1705,26 @@ func TestGenParameter_Issue1392(t *testing.T) {
 		}
 	}
 }
+
+func TestGenParameter_Issue1513(t *testing.T) {
+	assert := assert.New(t)
+	var assertion = `r.SetBodyParam(o.Something)`
+	gen, err := opBuilderWithFlatten("put-enum", "../fixtures/bugs/1513/enums.yaml")
+	if assert.NoError(err) {
+		op, err := gen.MakeOperation()
+		if assert.NoError(err) {
+			buf := bytes.NewBuffer(nil)
+			opts := opts()
+			err := templates.MustGet("clientParameter").Execute(buf, op)
```

Comment:
I copied this block from an earlier test case, but not that clear what its doing. Is this valid for checking the generation code?

Reply:
These lines are just to sanity check that the template executes correctly;
Unfortunately we don't have a real integration test on the CI.
We just run codegen asserts (you have a lot of that in this test file). They are here to validate that the codegen looks as expected, but not that it actually works...